src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -170,9 +171,10 @@ internal void AppendMetadata(MethodBodyStreamEncoder methodBodyEncoder)
                 }
 
                 WriteCustomAttributes(typeBuilder._customAttributes, typeHandle);
-                WriteProperties(typeBuilder);
                 WriteFields(typeBuilder);
                 WriteMethods(typeBuilder._methodDefinitions, genericParams, methodBodyEncoder);
+                WriteProperties(typeBuilder);
```

Comment:
Why the order change for properties? Does C# emit this way?

Reply:
nope, this order doesn't really matter as accessor methods handles are generated on type creation